platform/security/certificate/security-crl-generator/src/main/java/org/codice/ddf/security/crl/generator/CrlGenerator.java

```diff
@@ -0,0 +1,315 @@
+    Properties properties =
+        PROPERTIES_LOADER.loadPropertiesWithoutSystemPropertySubstitution(propertyFilePath, null);
+    properties.put(CRL_PROPERTY_KEY, localCrlPath);
+    try (OutputStream outputStream = new FileOutputStream(propertyFilePath)) {
+      properties.store(outputStream, null);
+    } catch (IOException e) {
+      LOGGER.warn(
+          "Unable to add the {} property to the property file {}.",
+          CRL_PROPERTY_KEY,
+          propertyFilePath);
+    }
+  }
+
+  /**
+   * Removes the org.apache.ws.security.crypto.merlin.x509crl.file property in the
+   * signature.properties and encryption.properties files.
+   */
+  @VisibleForTesting
+  void removeCrlFileLocationInPropertiesFile() {
+    removeProperty(issuerSignaturePropertiesLocation);
+    removeProperty(issuerEncryptionPropertiesLocation);
+    removeProperty(serverSignaturePropertiesLocation);
+    removeProperty(serverEncryptionPropertiesLocation);
+
+    SecurityLogger.audit(
+        "Removing the {} property from signature and encryption properties.", CRL_PROPERTY_KEY);
+  }
+
+  /**
+   * Removes the CRL file location from the given property file.
+   *
+   * @param propertyFilePath - Property file path
+   */
+  private void removeProperty(String propertyFilePath) {
+    Properties properties =
+        PROPERTIES_LOADER.loadPropertiesWithoutSystemPropertySubstitution(propertyFilePath, null);
+    properties.remove(CRL_PROPERTY_KEY);
+    try (OutputStream outputStream = new FileOutputStream(propertyFilePath)) {
+      properties.store(outputStream, null);
+    } catch (IOException e) {
+      LOGGER.warn(
+          "Unable to remove the {} property to the property file {}.",
+          CRL_PROPERTY_KEY,
+          propertyFilePath);
+    }
+  }
+
+  /**
+   * Sets the URL to download the CRL from and starts a task to download the CRL every 30 minutes.
+   *
+   * @param crlLocationUrl - CRL's URL location
+   */
+  public synchronized void setCrlLocationUrl(String crlLocationUrl) {
+    this.crlLocationUrl = crlLocationUrl;
+    if (handle != null) {
+      handle.cancel(true);
+    }
+
+    handle =
+        scheduler.scheduleAtFixedRate(this, INITIAL_DELAY, SCHEDULER_INTERVAL, TimeUnit.MINUTES);
+  }
+
+  /**
+   * Enables or disables the CRL download. If it's disabled, starts a task to remove the CRL file
+   * path from property files.
+   *
+   * @param crlByUrlEnabled - whether the feature is enabled or not
+   */
+  public synchronized void setCrlByUrlEnabled(boolean crlByUrlEnabled) {
+    if (this.crlByUrlEnabled && !crlByUrlEnabled) {
+      scheduler.schedule(
+          this::removeCrlFileLocationInPropertiesFile, INITIAL_DELAY, TimeUnit.MINUTES);
+    }
+
+    this.crlByUrlEnabled = crlByUrlEnabled;
+  }
+
+  /**
+   * Posts and error message to the Admin Console.
+   *
+   * @param errorMessage - The reason for the error.
+   */
+  private void postErrorEvent(String errorMessage) {
+    String title =
```

Comment:
✏️ Seems like we should probably add a security audit on any changes to the CRL configuration.